### ISSUE TITLE
"Touchable child must either be native or forward setNativeProps to a native component" error on the #next branch

### ISSUE THREAD

**Issue by xavier-villelegier** [react-native-training/react-native-elements#407]
Hey guys,

### Description

I was trying to use the #next branch to avoid having the `selectionColor` on Android, as suggested [here](https://github.com/react-native-training/react-native-elements/issues/331#issuecomment-304726022).

But i'm having another issue which is:

> Touchable child must either be native or forward setNativeProps to a native component
![image](https://cloud.githubusercontent.com/assets/17592779/26827901/7f85687c-4abf-11e7-8a3b-43a287cdbfcb.png)

To be sure, I also tested your "example" in `react-native-elements/example` with Expo, and the result is the same.

If you dismiss this error, you will have `undefined is not an object (evaluating 'this.refs[UNDERLAY_REF].setNativeProps)` on ListItems.

Maybe it is because of this warning ?
> Warning: Stateless function components cannot be given refs. Attempts to access this ref will fail.
> Check the render method of `TouchableHighlight`.
>    in Icon (at ListItem.js:193)

I just have this issue using the ListItem component.

### Reproduction Steps and Sample Code

To reproduce this, just put a `ListItem` in the first `render`.

### Solution

I don't have this issue on RNE `0.12.2`, or on the master branch. (Same goes for the `example` project.) Does the #next branch is supposed to be stable ? Or should we always be based on a specific release ?

**Comment by binoy14** [comment]
@xavier-villelegier thanks for reporting this. Need to fix before we release

**Comment by Monte9** [comment]
Duplicate of #406. Thanks for reporting this issue @xavier-villelegier. 

As Binoy said, `next` is our release branch and so this is a High Priority bug. We should have a fix for this soon.

If you have any ideas, please feel free suggest them. 

/cc @raarts please join the discussion here.

**Comment by Monte9** [comment]
Ok, I ran into this problem at work as well today and figure it out. This commit fixes it: https://github.com/react-native-training/react-native-elements/commit/714804d3c42b699c6a40c4c9f4a696206a960870

I think it's related to [this issue](https://github.com/facebook/react-native/issues/1040#issuecomment-96745870) and apparently, it was only fixed for `TouchableOpacity`.

So by using `TouchableOpacity` in place of `TouchableHighlight`, I was able to get rid of the error.
